helper/trace/debug_trace_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package trace
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestStartDebugTrace(t *testing.T) {
+	t.Run("error_on_non_existent_dir", func(t *testing.T) {
+		_, _, err := StartDebugTrace("non-existent-dir", "filePrefix")
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "does not exist")
+	})
+
+	t.Run("error_on_non_dir", func(t *testing.T) {
+		f, err := os.CreateTemp("", "")
+		require.NoError(t, err)
+		require.NoError(t, f.Close())
+		_, _, err = StartDebugTrace(f.Name(), "")
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "is not a directory")
+	})
+
+	t.Run("error_on_failed_to_create_trace_file", func(t *testing.T) {
+		noWriteFolder := filepath.Join(os.TempDir(), "no-write-permissions")
```

Comment:
nit: can use `t.TempDir()` to ensure that the directory gets cleaned up when the test exists